app/frontend/javascript/pool-xp-tube-panel/index.js

```diff
@@ -0,0 +1,66 @@
+/* eslint no-console: 0 */
+
+import Vue from 'vue'
+import { BootstrapVue, BootstrapVueIcons } from 'bootstrap-vue'
+import 'bootstrap/dist/css/bootstrap.css'
+import 'bootstrap-vue/dist/bootstrap-vue.css'
+import cookieJar from '@/javascript/shared/cookieJar.js'
+import PoolXPTubeSubmitPanel from './components/PoolXPTubeSubmitPanel.vue'
+
+Vue.use(BootstrapVue)
+Vue.use(BootstrapVueIcons)
+document.addEventListener('DOMContentLoaded', async () => {
+  /*
+   * As we add more components to this page we should
+   * consider switching to proper components and custom tags.
+   * Ran into a problems as I tried to do this at this stage:
+   * 1 - Vue needs to compile the template (ie. our HTML) on the fly
+   #     which means we import a different version of vue above.
+   #     import Vue from 'vue/dist/vue.esm'
+   #     This is slower, and generally recommended against.
+   # 2 - Things didn't appear to be as straight forward as I
+   #     had hoped. I *think* this was because I began wrestling
+   #     vue's expectations with regards to single page applications
+   # 3 - Vue does NOT like our existing templates. The script tags
+   #     seem to upset it.
+   # In general it looks like this is something we should consider
+   # once the majority of our components are vue based.
+   */
```

Comment:
Just noticing this comment is replicated a few times across the codebase now! Wonder if we should create a story / come up with a strategy for how we use Vue in Limber?

Reply:
Yes, this is what I mentioned in one of our previous discussions. think we could optimize how Vue components are loaded by encapsulating some of the generic structure and delegating certain responsibilities to helpers -perhaps.

Reply:
Do you think you could put some of your thoughts down in a story? With a bit more detail? When you feel you have time.

Reply:
Sure, I will create a story. 👍

Reply:
@KatyTaylor Story added https://github.com/sanger/limber/issues/2033

Reply:
Brilliant thank you